server/apps/on_post_created.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package apps
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+type PostCreatedNotification struct {
+	SubscriptionID SubscriptionID
+	Subject        SubscriptionSubject
+	ChannelID      string
+	ParentID       string
+	PostID         string
+	RootID         string
+	UserID         string
+	// TeamID         string
+	Expanded *Expanded
+}
+
+// OnUserHasBeenCreated sends a change notification when a new user has
+// joined a team.
+func (p *proxy) OnPostHasBeenCreated(ctx *plugin.Context, post *model.Post) {
+	subs, err := p.Subscriptions.GetSubscriptionsForChannelOrTeam(SubjectPostCreated, "")
+	if err != nil {
+		// p.Logger.Debugf("OnPostHasBeenCreated: failed to get subscriptions: %s %s: ",
+		// 	SubjectPostCreated, user.UserId, err)
+		return
+	}
+	if len(subs) == 0 {
+		return
+	}
+
+	expander := NewExpander(p.mm, p.configurator)
```

Comment:
1/5 At least this loop can be worked into an `ExpandNotification`-ish generic function